### PR TITLE
initial git post

### PR DESCRIPTION
**Pull request by gotoplanb**
I need some :eyes: on this post for comments.

@rgoff Can you read and comment?

Things to do before merging:

* [x] updated diagram showing patch and omitting hotfix branches. (created #16 for this improvement)

### CODE REVIEW

File: _posts/2015-03-23-beyond-git-flow.markdown [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+#### Creating the branch
+
+Every new feature should be developed in a **feature** branch. We  name the branch to include the creator's JIRA username and the JIRA ticket ID that covers this work (e.g. `feature/dstanton/MOB-123`). You should immediately publish your branch to GitHub, so that the team knows work is in progress. Publishing also provide an event JIRA can use to automatically assign the ticket to you and move the ticket to in-progress status. There should be a separate **feature** branch for each JIRA user story. 
+
+If a story is big, and multiple developers are contributing pieces, we should keep a single feature branch for that user story, and all developers should be pulling frequently. If a user story is so big that more than two developers are working, and pulling before push is inadequate to keep from stepping on each other, the team probably should split the large user story into multiple smaller user stories.
+
+#### Merging the branch
+
+When the ticket is complete, the developer should run `git rebase develop` before pushing the completed code to the feature branch. Rebasing helps us minimize merge conflicts.
```
Comment (jsposato) [inline _posts/2015-03-23-beyond-git-flow.markdown]: I would add that develop should be pull from origin first to make sure the rebase is against the most current version of develop

File: _posts/2015-03-23-beyond-git-flow.markdown [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+Commits to `develop` should be shortly available either as QA-ready binaries delivered through Apperian EASE (for iOS and Android) or deployed to a staging server (JavaScript and Java).
+
+### On-demand builds
+
+For projects that have minimal integration testing for whatever reason, we typically have another Jenkins job that builds on-demand from `develop`. We use this alternative approach with the construction team needs to smoke-test `develop` before sending to QA.
+
+## Release branches
+
+Upon sprint completion, the tech lead will create a **release** branch from `develop`. We like to follow [semantic versioning](http://semver.org/) and limit our release branches to specifying the minor version (e.g. `release/1.0`).
```
Comment (venksmob) [inline _posts/2015-03-23-beyond-git-flow.markdown]: What is the advantage of creating a Release branch upon sprint completion? If it's for tracking purposes, could we accomplish the same using Tagging?

File: _posts/2015-03-23-beyond-git-flow.markdown [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+## Feature branches
+
+All new work is done in a **feature** branch created from `develop`. There are multiple types of *features* branches, and naming convention changes slightly.
+
+### A new feature
+
+#### Creating the branch
+
+Every new feature should be developed in a **feature** branch. We  name the branch to include the creator's JIRA username and the JIRA ticket ID that covers this work (e.g. `feature/dstanton/MOB-123`). You should immediately publish your branch to GitHub, so that the team knows work is in progress. Publishing also provide an event JIRA can use to automatically assign the ticket to you and move the ticket to in-progress status. There should be a separate **feature** branch for each JIRA user story.
```
Comment (venksmob) [inline _posts/2015-03-23-beyond-git-flow.markdown]: You'll have to show me how I use the publishing for Jira to automatically assign the ticket to the author and move the ticket to in-progress status. That'd be great.

File: _posts/2015-03-23-beyond-git-flow.markdown [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+## git-flow is awesome but not a standard.
+
+Mobiquity follows the git-flow pattern described by [Vincent Driessen back in 2010](http://nvie.com/posts/a-successful-git-branching-model/). This pattern is widely used though naming details vary between organizations. I'm not going to enumerate the different naming schemes others use successfully. Instead, I  want to describe how Mobiquity names branches and merges.
+
+Open the [original git-flow article](http://nvie.com/posts/a-successful-git-branching-model/) so that you have the branching diagram for reference. Perhaps we should redraw branching diagram based on our internal conventions, but that is a task for a different day.
+
+## Feature branches
+
+All new work is done in a **feature** branch created from `develop`. There are multiple types of *features* branches, and naming convention changes slightly.
```
Comment (venksmob) [inline _posts/2015-03-23-beyond-git-flow.markdown]: What are the other naming conventions? I could not find the other examples in this article.

File: _posts/2015-03-23-beyond-git-flow.markdown (hunk 3 above)
Comment (venksmob) [inline _posts/2015-03-23-beyond-git-flow.markdown]: Sometimes a reviewer is unavailable to merge PRs. If a developer issues a PR and then needs to work on another feature or user story that is dependent on work done in the previous feature, can you write something about the work-flow that's involved then.

**Comment by rgoff** [comment]
Mobiquity development teams have used ad-hoc conventions in the past when concurrent work happens in the same project. This has included cases where two independent repositories existed for the same app. While this did cleanly separate the work, thereby protecting the new development effort from the maintenance fixes, it made for a difficult merge process at the conclusion of the maintenance cycle.

The specific instance case occurs when a **build** team is working on version 2.0, while at the same time a **support** team is working on a maintenance or hotfix release. This is one rationale behind the creation of a `release/1.1` branch, so that a support team can develop against a stable commit independent of ongoing work in the `develop` branch.

Since release branch commits could contain bug fixes, they could have PRs against both `release/1.1` and `develop` branches. The build team would decide each `develop` PR on a case-by-case basis. Enhancements, on the other hand, might be superseded by features under development by the build team. PRs would still be issued, but it seems likely that the build team would elect not to commit these enhancements back into the `develop` branch.

File: _posts/2015-03-23-beyond-git-flow.markdown [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+A **bug** branch lives in the same part of branching diagram as a **feature** branch and uses the same naming convention (e.g. `bug/dstanton/MOB-321`). We use the **bug** prefixing so that reviewer spends a bit of extra time on checking for regressions in related functionality. Otherwise, there isn't much difference between a **bug** branch and a **feature** branch.
+
+### Patch branch
+
+**Patch** branches live between `develop` and **release** branches. A **patch** branch is used when we have to make a fix against a **release** branch. For example, we may have a `release/3.6` branch that needs a fix. A ticket in JIRA exists that specifies the fix version as 3.6 and hence the developer creates a **patch** branch from the stated **release** branch instead using `develop` as the base branch. The naming convention stays the same (e.g. `patch/dstanton/MOB-444`).
+
+When dev is complete, two pull requests are created:
+
+1. From the **patch** branch into the relevant **release** branch
+2. From the **patch** branch into `develop`
```
Comment (venksmob) [inline _posts/2015-03-23-beyond-git-flow.markdown]: I am writing an article about the process/work-flow that is to be followed when doing the merging into release and develop (if you would like to crosslink)

**Comment by gotoplanb** [comment]
I'm just running out of time to draw this diagram, so i'll just merge and add the diagram as a separate issue for later since it's a bonus but not really required to understand the concepts.

**Pull request merged**